Bottom: 157453956f51de57da36948e98e1a62f1078fc17
Top:    25ee342b5843401aab39b272a1afe5d5bf756c7c
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-07-25 01:58:28 +0100

Expose draw.c internal drawing APIs


---

diff --git a/src/draw.c b/src/draw.c
index fa9e7f8..05ebad2 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -87,10 +87,7 @@ static void DrawPPV (int group, const BoxType *);
 static int DrawLayerGroup (int, const BoxType *);
 static void AddPart (void *);
 static void DrawEMark (ElementTypePtr, LocationType, LocationType, bool);
-static void DrawMask (int side, BoxType *);
-static void DrawPaste (int side, BoxType *);
 static void DrawRats (BoxType *);
-static void DrawSilk (int side, const BoxType *);
 
 
 /*---------------------------------------------------------------------------
@@ -652,7 +649,7 @@ pad_mask_callback (const BoxType * b, void *cl)
  * Draws silk layer.
  */
 
-static void
+void
 DrawSilk (int side, const BoxType * drawn_area)
 {
 #if 0
@@ -711,7 +708,7 @@ DrawMaskBoardArea (int mask_type, BoxType *screen)
 /* ---------------------------------------------------------------------------
  * draws solder mask layer - this will cover nearly everything
  */
-static void
+void
 DrawMask (int side, BoxType * screen)
 {
   int thin = TEST_FLAG(THINDRAWFLAG, PCB) || TEST_FLAG(THINDRAWPOLYFLAG, PCB);
@@ -740,7 +737,7 @@ DrawMask (int side, BoxType * screen)
 /* ---------------------------------------------------------------------------
  * draws solder paste layer for a given side of the board
  */
-static void
+void
 DrawPaste (int side, BoxType *drawn_area)
 {
   gui->set_color (Output.fgGC, PCB->ElementColor);
diff --git a/src/draw.h b/src/draw.h
index cf324a5..5224adc 100644
--- a/src/draw.h
+++ b/src/draw.h
@@ -71,6 +71,10 @@ void EraseElementName (ElementTypePtr);
 void EraseObject (int, void *, void *);
 void LoadBackgroundImage (char *);
 
+void DrawMask (int side, BoxType *);
+void DrawSilk (int side, const BoxType *);
+void DrawPaste (int side, BoxType *);
+
 /*GdkDrawable *draw_get_current_drawable(void);*/
 
 #endif
